### ISSUE TITLE
Another issue with invalid Lambda function's code path (adding to #1270)

### ISSUE THREAD

**Issue by adrians5j** [pulumi/pulumi-aws#1274]
## Pulumi versions
`@pulumi/aws` => `v3.5.0`
`@pulumi/pulumi` => `2.11.2`

Pulumi CLI: `2.11.2`

## Problematic Behaviour
Adding to #1270, today I experienced a similar error, although I think this is much more serious I believe.

In the screenshot below, I essentially have two functions (classes just encapsulate cloud resources, don't let those confuse you):

![image](https://user-images.githubusercontent.com/5121148/102721829-875b5180-42fd-11eb-9b1f-36f9594c37c7.png)

As we can see, I'm passing `PB_RENDER_FUNCTION` as the environment variable for the functions in `Graphql` class.

Now, please let me explain what I experienced today, and it kinda concerns me.

In the `PrerenderingService` class, I actually had the incorrect path specified:

![image](https://user-images.githubusercontent.com/5121148/102721930-15373c80-42fe-11eb-9fa9-acd9bedf40d4.png)

So, this path `"./code/pageBuilder/render/build"` is actually incorrect.

And then what happened...

Not knowing that the path was incorrect, when I tried to deploy the stack, I received the following error:

![image](https://user-images.githubusercontent.com/5121148/102721972-592a4180-42fe-11eb-8615-22335f8ebe9f.png)

First of all, as you can expect, this doesn't tell me at all what is wrong.  OK, it's not a big deal. I guess again some error reporting that needs to be improved. Not a biggie, I found the incorrect path eventually.

But, once I finally figured the path is incorrect, I tried to redeploy.

 And this is the part that concerns me...

Instead of redeploying, Pulumi started to create a brand new stack. Instead of updating the existing cloud resources.

![image](https://user-images.githubusercontent.com/5121148/102722048-cccc4e80-42fe-11eb-82ec-b0f97116463c.png)

Now, this is definitely something I didn't expect. Why did the Pulumi framework "loose the connection" with my current state files (stored locally), I don't know.

Kinda sounds scary that Pulumi "lost the connection" with my existing state files and now I have to deploy everything from scratch.

I'm not complaining don't get me wrong. I just wanted to bring attention to this issue. 

I guess it's possible to maybe tell the Pulumi to stop and "connect" with my current state files. Of course, I'm not ruling out the possibility that I'm maybe using something wrong on my end. If so, sorry about that :)

--- 
P.S. What is also interesting... if I tried to deploy the stack, with the `PrerenderingService` class instantiated (meaning the functions inside of the class will be deployed), no errors pop up upon the stack deployment. 
![image](https://user-images.githubusercontent.com/5121148/102722213-e9b55180-42ff-11eb-9878-83bb632fe495.png)

Only once I removed the comment and actually tried to pass the `PB_RENDER_FUNCTION` env variable value, the error above would show.

**Comment by leezen** [comment]
Thanks for reporting this. Do you have to contents of the state file after you received the "failed to save snapshot" error? I would be interested in seeing that as the only way I could see the next thing happening after that is if the state file itself has incorrect information in it or the state file itself is not in the correct location after that error.

**Comment by lukehoban** [comment]
I don't see any indication from the screenshots that this is related to lambda function serialization, and as noted in #1270, that issue itself has been addressed since this was opened.

The errors shown above seem to suggest some failure to write to state storage on your local machine.  There have been some fixes in that area too, but it's hard to tell just from the above what exactly the problem was, or whether it was reproducible.

I'm going to close this as Not Repro for now, and if there are new issues related to state storage writes, those can be opened in pulumi/pulumi with repro steps.
